tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,352 @@ use strict;
     wait_screen_change { send_key 'alt-o' };
-
     # cancel trusted domain configuration
     wait_screen_change { send_key 'alt-c' };
 
-}
-
-sub samba_ldap_config {
-    # swith to LDAP Settings
-    send_key 'alt-l';
-    assert_screen 'yast2_samba-server_ldap-settings';
-    wait_screen_change { send_key 'alt-b' };
-    assert_screen 'yast2_samba-server_ldap_value_rewritten';
-    wait_screen_change { send_key 'alt-y' };
-    wait_screen_change { send_key 'alt-e' };
-    type_string 'ldap://localhost:389';
-
-    # set admin password and search base dn
-    wait_screen_change { send_key 'alt-a' };
-    type_string 'cn=admin,dc=qa,dc=suse,dc=de';
-    wait_screen_change { send_key 'alt-p' };
-    type_string 'testing';
-    wait_screen_change { send_key 'alt-g' };
-    type_string 'testing';
-    wait_screen_change { send_key 'alt-n' };
-    type_string 'dc=qa, dc=suse, dc=de';
-
-    # check advanced settings befor run test connection to ldap server
-    wait_screen_change { send_key 'alt-v' };
-    send_key 'ret';
-
-    # enter expert ldap settings
-    assert_screen 'yast2_samba-server_ldap_advanced_expert_settings';
-
-    # change replication sleep and time-out
-    wait_screen_change { send_key 'alt-p' };
-    for (1 .. 10) { send_key 'down'; }
-    wait_screen_change { send_key 'alt-t' };
-    for (1 .. 2) { send_key 'up'; }
-
-    # change to not use SSL or TLS
-    wait_screen_change { send_key 'alt-u' };
-    for (1 .. 2) { send_key 'up'; }
-    assert_screen 'yast2_samba-server_ldap_advanced_expert_settings_not-use-ssl';
-    wait_screen_change { send_key 'ret' };
-    wait_screen_change { send_key 'alt-o' };
-
-    # now run test connection
-    send_key 'alt-t';
-    assert_screen 'yast2_samba-server_ldap_test-connection';
-    wait_screen_change { send_key 'alt-o' };
+    if ((is_sle(">=15")) || (is_leap(">=15.0")) || is_tumbleweed) {
+        record_soft_failure "bsc#1088152";
+        # _setup_ldap_in_samba; commented due to above bug
+    }
+    else {
+        _setup_ldap_in_samba;
+    }
 
     # finally, close with OK
     send_key 'alt-o';
-
     wait_serial('yast2-samba-server-status-0', 60) || die "'yast2 samba-server' didn't finish";
+
 }
 
-sub gui_new {
-    assert_screen 'yast2_samba-workgroup';
-    send_key 'alt-n';
-    assert_screen([qw(yast2_samba-firewall_active yast2_still_susefirewall2)]);
+sub setup_yast2_auth_server {
+    # workaround kernel message floating over console
+    assert_script_run "dmesg -n 4";
+
+    # check network at first
+    assert_script_run("if ! systemctl -q is-active network; then systemctl -q start network; fi");
+
+    script_run("yast2 auth-server; echo yast2-auth-server-status-\$? > /dev/$serialdev ", 0);
+
+    # check ldap server configuration started
+    assert_screen([qw(yast2_ldap_configuration_startup yast2_still_susefirewall2)], 60);
     if (match_has_tag 'yast2_still_susefirewall2') {
         record_soft_failure 'bsc#1064405';
```

Comment:
I know this about the only code that you did not have to change :D But I would appreciate if you remove that block and the `record_soft_failure` completely as [bsc#1064405](https://bugzilla.suse.com/show_bug.cgi?id=1064405) is VERIFIED FIXED